mmds/src/lib.rs

```diff
@@ -22,6 +21,32 @@ lazy_static! {
     pub static ref MMDS: Arc<Mutex<Mmds>> = Arc::new(Mutex::new(Mmds::default()));
 }
 
+/// Patch provided JSON document (given as `serde_json::Value`) in-place with JSON Merge Patch
+/// [RFC 7396](https://tools.ietf.org/html/rfc7396).
+pub fn json_patch(target: &mut Value, patch: &Value) {
+    if patch.is_object() {
+        if !target.is_object() {
+            // Replace target with an empty json so we can recursively copy patch values.
+            *target = Value::Object(Map::new());
+        }
+        let doc = target.as_object_mut().unwrap();
```

Comment:
We are trying to avoid using unwrap's in our code and when we do we add a comment specifying why it is safe to do so (i.e. will never cause a crash).
Does the validations that target and patch go through before calling this function (i.e check_data_store_initialized and Mmds::check_data_valid, respectively) ensure that these unwraps can never cause a crash?
[Later Edit]: I see now you are making sure that target and patch are objects.  I'd still add a comment before the unwraps saying:
// This is safe since we make sure `target` and `patch` are objects.